policy/modules/apps/screen.if

```diff
@@ -48,31 +49,31 @@ template(`screen_role_template',`
 
 	dontaudit $1_screen_t self:capability sys_tty_config;
 
-	domtrans_pattern($3, screen_exec_t, $1_screen_t)
+	domtrans_pattern($1_user_exec_domain, screen_exec_t, $1_screen_t)
```

Comment:
This won't work for *_role() interfaces that don't use a prefix. The $1_user_exec_domain needs to be passed in as $1.

Reply:
>The $1_user_exec_domain needs to be passed in as $1.

The problem with this is that some `*_role()` interfaces associate an attribute with `$1`, and since `$1_user_exec_domain` is an attribute this will never work, which is my rationale for reworking all `*_role()` interfaces to take `$3` as the user prefix, or maybe just the user domain (i.e. `$3_t`).

Reply:
hmm, maybe we should expand it to be *_role([prefix], domain, exec_domain_attr, role) to cover both cases?

Reply:
I think that would be a good solution. As long as we can ensure that `*_role()` interfaces can have the prefix, the domain, and the attribute, then there won't be any issues with creating the required rules. Updating this PR soon so we can see what that will look like.